server/src/main/java/org/opensearch/search/pipeline/SearchPipelineMetrics.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
```

Comment:
`SearchPipelineMetrics` is very similar implementation of `IngestMetric`[1]. Lets consolidate the logic with a base class? 

[1] https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/ingest/IngestMetric.java

Reply:
I think they can actually be one class used in both places. I'll make that change.